### DISCUSSION TITLE
Bump package version only on build outputs and not in source

### DISCUSSION THREAD

**Discussion by SebastienGllmt** [verdaccio/verdaccio#4866] [Ideas]
**Is your feature request related to a problem?**

To work properly with package manages (npm/pnpm/yarn), new code for a package should always be accompanies with a new version number

However, I use Verdaccio to quickly iterate on my package locally and therefore I do **NOT** want a new version number for my package every single time I want to do integration tests of my work. Notably, solutions that `new Date().getTime()` to my package version every build also are very annoying as they cause a bunch of git diffs which I have to remember to undo when I publish my to avoid creating a bunch of git conflicts with the rest of my team.

**Describe the solution you'd like**
A way to bump the version number of my package (ex: appending `new Date().getTime()`) but ONLY in Verdaccio's view of the package and not actually on disk

**Describe alternatives you've considered**
Changes could be made in the package managers to alleviate this instead of making a code change in Verdaccio. See https://github.com/pnpm/pnpm/issues/8578

**Comment by mbtools** [comment]
I guess, you tried `npm unpublish` and `npm publish` on the same version again, and  `ppm` didn't like it (since the checksum changes). That's too bad.

You could try creating a [middleware plugin](https://www.npmjs.com/package/verdaccio-npm-urls) that intercepts the publish request which contains `package.json` in the body (it's a `PUT` to `https://localhost:4873/{package}`). Adjust the version as you like and verdaccio should store it like that.
